lib/irb.rb

```diff
@@ -513,34 +514,38 @@ def run(conf = IRB.conf)
     # The lexer used by this irb session
     attr_accessor :scanner
 
-    # Evaluates input for this session.
-    def eval_input
-      @scanner.set_prompt do
-        |ltype, indent, continue, line_no|
-        if ltype
-          f = @context.prompt_s
-        elsif continue
-          f = @context.prompt_c
-        else
-          f = @context.prompt_i
-        end
-        f = "" unless f
-        if @context.prompting?
-          @context.io.prompt = p = prompt(f, ltype, indent, line_no)
-        else
-          @context.io.prompt = p = ""
-        end
-        if @context.auto_indent_mode and !@context.io.respond_to?(:auto_indent)
-          unless ltype
-            prompt_i = @context.prompt_i.nil? ? "" : @context.prompt_i
-            ind = prompt(prompt_i, ltype, indent, line_no)[/.*\z/].size +
-              indent * 2 - p.size
-            @context.io.prompt = p + " " * ind if ind > 0
-          end
+    def prompt(opens, continue, line_offset)
```

Comment:
Can we call it `generate_prompt` or `generate_prompt_string` to make its action more clear?